services/m3aggregator/config/aggregator.go

```diff
@@ -282,11 +286,16 @@ func (c *AggregatorConfiguration) NewAggregatorOptions(
 	entryPool.Init(func() *aggregator.Entry { return aggregator.NewEntry(nil, opts) })
 
 	// Set buffered encoder pool.
+	// NB(xichen): we preallocate a bit over the maximum flush size as a safety measure
```

Comment:
Perhaps move this line to before the line where we determine the `intialBufferSize` just so its a littler clearer that that is the line its referring to?